flink-formats/flink-csv/src/test/java/org/apache/flink/formats/csv/TableCsvFormatITCase.java

```diff
@@ -18,32 +18,78 @@
 
 package org.apache.flink.formats.csv;
 
+import org.apache.flink.connector.file.table.FileSystemConnectorOptions;
+import org.apache.flink.connector.file.table.FileSystemTableFactory;
 import org.apache.flink.formats.common.TimeFormats;
+import org.apache.flink.table.api.EnvironmentSettings;
+import org.apache.flink.table.api.Schema;
+import org.apache.flink.table.api.TableDescriptor;
+import org.apache.flink.table.api.TableEnvironment;
+import org.apache.flink.table.planner.factories.TestValuesTableFactory;
 import org.apache.flink.table.planner.runtime.utils.TestData;
 import org.apache.flink.table.planner.utils.JavaScalaConversionUtil;
-import org.apache.flink.table.planner.utils.JsonPlanTestBase;
+import org.apache.flink.test.util.AbstractTestBase;
+import org.apache.flink.types.Row;
 
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
 import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import javax.annotation.Nullable;
 
 import java.io.File;
 import java.io.IOException;
 import java.nio.file.Files;
+import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 
+import static org.apache.flink.table.api.DataTypes.BIGINT;
+import static org.apache.flink.table.api.DataTypes.INT;
+import static org.apache.flink.table.api.DataTypes.STRING;
+import static org.apache.flink.table.api.DataTypes.TIMESTAMP;
 import static org.apache.flink.table.utils.DateTimeUtils.toLocalDateTime;
+import static org.apache.flink.util.Preconditions.checkNotNull;
+import static org.assertj.core.api.Assertions.assertThat;
 
 /** Tests for the CSV file format. */
-public class TableCsvFormatITCase extends JsonPlanTestBase {
+public class TableCsvFormatITCase extends AbstractTestBase {
+
+    @Rule public ExpectedException exception = ExpectedException.none();
+
+    private TableEnvironment tableEnv;
+
+    @Before
+    public void setup() throws Exception {
+        tableEnv = TableEnvironment.create(EnvironmentSettings.inStreamingMode());
+    }
+
+    @After
+    public void after() {
+        TestValuesTableFactory.clearAllData();
+    }
 
     @Test
     public void testProjectPushDown() throws Exception {
         List<String> data = Arrays.asList("1,1,hi", "2,1,hello", "3,2,hello world");
-        createSourceTable("MyTable", data, "a bigint", "b int not null", "c varchar");
-        File sinkPath = createSinkTable("MySink", "a bigint", "c varchar");
+
+        Schema sourceSchema =
+                Schema.newBuilder()
+                        .column("a", BIGINT())
+                        .column("b", INT())
+                        .column("c", STRING())
+                        .build();
+
+        createSourceTable("MyTable", data, sourceSchema);
+
+        Schema sinkSchema = Schema.newBuilder().column("a", BIGINT()).column("c", STRING()).build();
+
+        File sinkPath = createSinkTable("MySink", sinkSchema);
```

Comment:
Can we just reuse sourceSchema here?

Reply:
This test checks that projection pushdown from the SQL engine works. Notice that after dropping `b` column from the sink schema, the returned elements are expected to also skip it.

Reply:
You're right, I overlooked the `b` part, sorry